### ISSUE TITLE
can't find out all subtitles 

### ISSUE THREAD

**Issue by justinluy** [morpheus65535/bazarr#915]
i am using zimuku for chinese subtitles, but seems it cannot scrap all subtitles. like this one
totally 5 subtitles are found on zimuku, the last one is english, the top one is subtitles for whole season, not sure if it can be scrapped. so at least 3 subtitles should be scrapped on bazarr, anyhow, only 2 were found which only 94% score while the the left 1 are almost 100% score.

![image](https://user-images.githubusercontent.com/63279919/78719573-3e828900-7978-11ea-805e-43663bbf0513.png)

![image](https://user-images.githubusercontent.com/63279919/78720015-0891d480-7979-11ea-9ce1-b79865408d51.png)

![image](https://user-images.githubusercontent.com/63279919/78720149-43940800-7979-11ea-8f61-5e4bb0fd7708.png)

can anyone help please? Thanks.

**Comment by morpheus65535** [comment]
Without a debug log (emptied first) reproducing the issue, there's nothing we can do.

**Comment by justinluy** [comment]
> Without a debug log (emptied first) reproducing the issue, there's nothing we can do.

sorry, here it is 

[bazarr.log](https://github.com/morpheus65535/bazarr/files/4447609/bazarr.log)

**Comment by morpheus65535** [comment]
On MArch 25th there's been fixes made to Zimuku provider in dev branch. I suggest you test it. From the look of your screenshot, I guess you've installed Bazarr through Windows Installer. If that's the case, you cannot switch to something else than master branch. You'll have to run it from source if you want to test before those fixes get merged to master branch sometime in the upcoming weeks.

**Comment by justinluy** [comment]
sorry, i am totally a newbie about these codes stuff. anyway, i installed from source as following
![image](https://user-images.githubusercontent.com/63279919/78741650-01d18480-79ae-11ea-9593-2a64822b6b9e.png)
i guess the problem you mentioned was it couldn't connect to zimuku provider and download. i installed through windows installer before and found this problem, then i switched to current version. took me a while to figure how to install though.

**Comment by morpheus65535** [comment]
This is the fork where the fixes have been developed. You should have installed from `morpheus/bazarr:development` as this fork won't be necessary updated by the contributor in the future. You can simply backup your data directory, delete bazarr directory and git clone again but from my repo. You'll be able to restore your data directory before starting Bazarr. You should be good with that.

**Comment by justinluy** [comment]
thank you very much for the prompt reply. sorry for this stupid question, how do i install from dev? i chose the development and clicked clone, still got the master one.

**Comment by morpheus65535** [comment]
You start by following the wiki: https://github.com/morpheus65535/bazarr/wiki/Installation-Running-from-Source

Then, when you got Bazarr running, you switch from master to dev at the bottom of Settings-->General page. You click save, go to System-->Tasks and force an "update from source on Github". Once the update is completed, you restart Bazarr.

**Comment by justinluy** [comment]
thank you, i got the dev version installed, still the same problem and now i got provider oserror. 
[bazarr (1).log](https://github.com/morpheus65535/bazarr/files/4448178/bazarr.1.log)

**Comment by morpheus65535** [comment]
There's nothing relevant in your log as `BAZARR All providers are throttled`.

**Comment by justinluy** [comment]
what does this mean? I am banned from zimuku?

**Comment by morpheus65535** [comment]
It just means that Bazarr isn't doing more request to zimuku to make sure you don't get banned. That's a safe guard to protect you from hammering the provider.

**Comment by justinluy** [comment]
ok I will wait to tomorrow. but I guess this problem still exists, it just can't scrap all subs for some reasons.

**Comment by morpheus65535** [comment]
It shouldn't throttle you for more than 10 minutes so you don't have to wait until tomorrow...

**Comment by justinluy** [comment]
I am still throttled.

**Comment by morpheus65535** [comment]
I've just tested Zimuku for a Chinese subtitles for the same episode and it works fine for me:
![image](https://user-images.githubusercontent.com/5130500/78846337-69e99000-79d9-11ea-9a40-e2cd4c41bbcd.png)


I don't know what's different for you but for sure I cannot reproduce.

**Comment by justinluy** [comment]
that's weird, I guess I will reinstall again.

**Comment by morpheus65535** [comment]
You can try to delete your data directory and start from scratch. In fact oserror is common when Bazarr cannot access the file (non-existant path, permission issue maybe).

**Comment by justinluy** [comment]
ok, i may found the problem. i used shortcut pointing to pythonw instead of python,wanted to start up with windows and show no window. any suggestion to hide the cmd window?
and back to my original issue, i now can get 3 chinese subtitles, anyhow, still one missing which is almost 100% match and that is i need. can you check?

![image](https://user-images.githubusercontent.com/63279919/78851304-92bb6500-7a6d-11ea-8a93-f628efb3c9a5.png)

**Comment by morpheus65535** [comment]
Read the wiki. I can't read a single word from your screenshot so it's kinda hard to help you.

**Comment by justinluy** [comment]
those chinese just says series title, season, sub language, sub release group, nothing important. i was wondering is this title too long to make it unreachable? as following one, bazarr can find it without problem. same sub release group, just without chinese character.
![image](https://user-images.githubusercontent.com/63279919/78852012-60ab0280-7a6f-11ea-9020-6c1409b90187.png)

**Comment by morpheus65535** [comment]
This one is for S10E12 so it's normal that it isn't returned as a result when searching for S10E15.

**Comment by justinluy** [comment]
yes i know. i mean this e12 and e15 are from same group (yyets) and perfectly match my file.
for e12 100% match 
for e15 it is The.Walking.Dead.S10E15.The.Tower.720p/1080p.AMZN.WEB-DL.DD+5.1.H.264-CasStudio
my file is The.Walking.Dead.S10E15.The.Tower.1080p.AMZN.WEB-DL.DD+5.1.H.264-CasStudio
only difference is that 720p. and this e15 subtitle can not be found.

**Comment by justinluy** [comment]
the result returned you tested was missing this identical one as well.

**Comment by morpheus65535** [comment]
@gyh1621 any idea why this subtitles isn't included in results?

**Comment by gyh1621** [comment]
Actually "zimuku" found this subtitle, but it is filtered out because the episode name extracted from subtitle by `guessit` not match the video file's name.

Let me explain in detail.

First of all, as you can see, the names of subtitles provided by the "zimuku" website are always mixed with Chinese characters, so I have to extract English names first. Here is the [function](https://github.com/morpheus65535/bazarr/blob/development/libs/subliminal_patch/providers/zimuku.py#L301). The function will extract the English name by finding the longest non-Chinese substring.

The missing subtitle's original name is
`行尸走肉 第10季第15集【本季终】.The.Walking.Dead.[WEB.1080P]中英文字幕【YYeTs字幕组 简繁英双语字幕】The.Walking.Dead.S10E15.The.Tower.720p/1080p.AMZN.WEB-DL.DD+5.1.H.264-CasStudio`
so the English name is
`The.Walking.Dead.S10E15.The.Tower.720p/1080p.AMZN.WEB-DL.DD+5.1.H.264-CasStudio`

Next, when coming to the matching step, `guessit` will give below result:
```
MatchesDict([('season', 10),
             ('episode', 15),
             ('screen_size', '1080p'),
             ('streaming_service', 'Amazon Prime'),
             ('source', 'Web'),
             ('audio_codec', 'Dolby Digital Plus'),
             ('audio_channels', '5.1'),
             ('title', 'H'),
             ('container', 'mkv'),
             ('mimetype', 'video/x-matroska'),
             ('type', 'episode')])
``` 
And it is missing the `title` infomation because the "720p/1080p" in the subtitle name...

I currently can't come up with a better solution to "format" original subtitle names.

**Comment by gyh1621** [comment]
I tested just letting `guessit` process original names and the results seem pretty good except the title parsed will contain Chinese characters. Maybe I can do the Chinese-filtering process after getting `guessit` result.

It's just a thought. I don't know how this will affect the overall results of `zimuku` provider. And I kind of busy lately, so temporarily I won't look into this furthermore ...

**Comment by morpheus65535** [comment]
@gyh1621 no problem, thanks mate for taking a look! I think that cleaning the title after guessit analyze would worth a try. When you'll get some free time, I would appreciate to get your help with that one. Thanks!

**Comment by justinluy** [comment]
thanks guys.

**Comment by morpheus65535** [comment]
@justinluy I've updated guessit in dev branch. You can take a look and report if it fix your issue?

**Comment by justinluy** [comment]
> @justinluy I've updated guessit in dev branch. You can take a look and report if it fix your issue?

the page looks like this now.
![Screenshot_20200525-183314_Chrome](https://user-images.githubusercontent.com/63279919/82785293-7016cb80-9eb6-11ea-8020-950766c8d966.jpg)

**Comment by morpheus65535** [comment]
Web browser cache problem.

**Comment by justinluy** [comment]
> Web browser cache problem.

i am using mobile phone? changed another browser, still the same.

**Comment by morpheus65535** [comment]
Your browser is preventing the download of css file so you're not getting the proper UI. Using a desktop browser, you can confirm if it works and if not, share the browser dev console output with me.

**Comment by justinluy** [comment]
just checked, version 0.8.4.4 unfortunately not fixed.

**Comment by morpheus65535** [comment]
I've made some improvement and can confirm that all subtitles that aren't assrt are now available. I've improved year matching as it was previously assumed that year was always a match even if provider was wrong. Can someone test it?

**Comment by justinluy** [comment]
just tested, seems still no luck.
![image](https://user-images.githubusercontent.com/63279919/87245016-dcb14e00-c495-11ea-908e-298d197c357b.png)
![image](https://user-images.githubusercontent.com/63279919/87245024-f6529580-c495-11ea-87de-0b0cf1884a46.png)

**Comment by morpheus65535** [comment]
By looking at the code, I figured out that the subtitles reposted from assrt are excluded. Could that be the case here?

**Comment by justinluy** [comment]
i didn't use assrt, i was using zimuku

**Comment by morpheus65535** [comment]
I know but Zimuku is reposting assrt subtitles. As we also have an assrt provider, we filter the subtitles returned by Zimuku that are in fact available on assrt. This is probably the "issue" you are having.

**Comment by justinluy** [comment]
hmmm probably.

**Comment by morpheus65535** [comment]
Unless you tell me otherwise, I would consider this issue closed.

**Comment by justinluy** [comment]
but i cannot use assrt. always shows 

<img width="893" alt="Annotation 2020-07-14 221503" src="https://user-images.githubusercontent.com/63279919/87414231-94259c00-c61f-11ea-8dfb-36803718d992.png">

**Comment by morpheus65535** [comment]
30 subtitles have been downloaded from assrt yesterday. I suggest you try the dev branch that include some fixes for assrt and open an new issue with a debug log if it still doesn't work.

**Comment by justinluy** [comment]
i think i am on dev branch?

![image](https://user-images.githubusercontent.com/63279919/87417247-faacb900-c623-11ea-8f8e-0ce2fbe50154.png)
![image](https://user-images.githubusercontent.com/63279919/87417276-0304f400-c624-11ea-9c2d-7d663390ba58.png)

**Comment by morpheus65535** [comment]
Yes

**Comment by justinluy** [comment]
thanks, i will open an new ticket about assrt issue.
